### DISCUSSION TITLE
pagImageView 使用的pag文件能提前预加载

### DISCUSSION THREAD

**Discussion by xyggs** [Tencent/libpag#1573] [Q&A] [answered]
【版本信息】
PAG SDK 版本 4.2.100
【平台信息】
 Android 原生 

【预期的表现】
在某个activity里 会加载10多个pagImageView, 初次加载卡顿明显，希望在demo 里面提供 预加载的实例

**Comment by domchen** [comment, marked as answer]
这个实际上非常的简单，只要几句话就描述清楚了：你从PAGFile创建一个PAGDecoder，创建的时候通过scale参数指定好分辨率，要匹配PAGImageView里会用到的实际尺寸，然后for循环调用 PAGDecoder的frameAtIndex方法，全都执行一遍就做完预加载了。

**Reply by xyggs** [reply]
噢，感谢~ 那有办法知道这个pagFile 是否缓存成功吗， 现在执行完了frameAtIndex，但不知道怎么判断是否缓存过了，所以还是每次在splash时都在调用这个，在测试机上耗时大概1420ms

**Reply by domchen** [reply]
现在的API里没有这块的能力，暂时没想到特别好的暴露方式，建议可以自己在业务上做个持久化的标识，缓存过了就不启动。

**Comment by yanyanyy** [comment]
这个会有PAG的交流群吗
预加载这样写是正确的吗
pagFileYearStart = PAGFile.Load(context.assets, "pag/new_year/new_year_start.pag")
PAGDecoder.Make(pagFileYearStart).frameAtIndex(0)

**Reply by kevingpqi123** [reply]
这部分你可以看下 pag 的源码， 如果需要预加载，PAGDecoder创建的尺寸不是 pag 文件的尺寸，和 View 的大小强相关，有一个缩放参数
